akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala

```diff
@@ -274,25 +301,29 @@ object Framing {
               nextPossibleMatch = buffer.size
               doParse()
             }
-          } else if (possibleMatchPos + separatorBytes.size > buffer.size) {
-            // We have found a possible match (we found the first character of the terminator
-            // sequence) but we don't have yet enough bytes. We remember the position to
-            // retry from next time.
-            nextPossibleMatch = possibleMatchPos
-            doParse()
-          } else if (buffer.slice(possibleMatchPos, possibleMatchPos + separatorBytes.size) == separatorBytes) {
-            // Found a match, mark start and end position and iterate if possible
-            indices += (previous -> possibleMatchPos)
-            nextPossibleMatch = possibleMatchPos + separatorBytes.size
-            if (nextPossibleMatch == buffer.size || indices.isFull) {
-              doParse()
-            } else {
-              searchIndices()
-            }
           } else {
-            // possibleMatchPos was not actually a match
-            nextPossibleMatch += 1
-            searchIndices()
+            val delimiters = delimitersByFirstByte(buffer(possibleMatchPos))
+            delimiters.find(delimiter => buffer.startsWith(delimiter, offset = possibleMatchPos)) match {
```

Comment:
This stage was highly optimized to not do unnecessary allocations, this will allocate a function for the `find` parameter and a `Some` wrapper for the found case, so we will likely want to avoid using find.